resilience4j-core/src/main/java/io/github/resilience4j/core/EventProcessor.java

```diff
@@ -25,16 +25,16 @@
 
 public class EventProcessor<T> implements EventPublisher<T> {
 
-    protected volatile boolean consumerRegistered;
-    @Nullable private volatile EventConsumer<T> onEventConsumer;
+    private boolean consumerRegistered;
+    @Nullable private EventConsumer<T> onEventConsumer;
     private ConcurrentMap<Class<? extends T>, EventConsumer<Object>> eventConsumers = new ConcurrentHashMap<>();
 
     public boolean hasConsumers(){
         return consumerRegistered;
     }
 
     @SuppressWarnings("unchecked")
-    public <E extends T> void registerConsumer(Class<? extends E> eventType, EventConsumer<E> eventConsumer){
+    public synchronized <E extends T> void registerConsumer(Class<? extends E> eventType, EventConsumer<E> eventConsumer){
```

Comment:
why u need it synchronized ?

Reply:
It was a SonarCloud warning. 
The method `onEvent` changes two fields, but both fields are only volatile. This can cause problems.
The method `registerConsumer` registerConstumer also updates two fields.